### PR TITLE
Fix(security): #MVS-107 APIs security for EleveController part2

### PR DESCRIPTION
**Pull request by Gregprod33**
## Describe your changes
Fix security on routes :
@Get("/enseignants")  => Removed because unused
@Get("/eleves") => update neo4j request to remove birthDate from query response and resource filter to check structure.
@Get("/users") => Removed because unused
@Get("/cycle/eleve/:idClasse") => SecuredAction authenticated actually fits well for this route
@Get("/structures/:structureId/students") => Resource filter to check structure.
## Issue ticket number and link
[#MVS-107](https://jira.support-ent.fr/browse/MVS-107)
## Checklist before requesting a review (magic string, indentation, comment/documentation...)
Only one route to test :
**@Get("/structures/:structureId/students")** => Vie scolaire => Etablissement => Trombinoscope
- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

**Comment by lapeyronnie** [comment]
Peux-tu ajouter, stp, les tests à effectuer pour tester la PR.

File: src/main/java/fr/openent/viescolaire/service/impl/DefaultEleveService.java [hunk 1]
```diff
@@ -158,7 +158,7 @@ public void getInfoEleve(String[] idEleves, String idEtablissement, Handler<Eith
         StringBuilder returning = new StringBuilder()
                 .append(" RETURN u.id as idEleve, u.firstName as firstName, u.lastName as lastName, u.ine as ine, ")
                 .append(" u.deleteDate,c.id as idClasse, c.name as classeName, s.id as idEtablissement, ")
-                .append(" u.birthDate as birthDate, u.level as level, c.externalId as externalId, ");
+                .append(" u.level as level, c.externalId as externalId, ");
```
Comment (lapeyronnie) [inline src/main/java/fr/openent/viescolaire/service/impl/DefaultEleveService.java:161]: ne pas supprimer, la date de naissance l'utilisateur : bécessaire pour le bulletin

File: src/main/java/fr/openent/viescolaire/controller/EleveController.java [hunk 2]
```diff
@@ -213,8 +190,8 @@ public void getAppreciationDevoirEleve(final HttpServerRequest request) {
 
     @Get("/structures/:structureId/students")
     @ApiDoc("Get list of students from a structure")
-    @ResourceFilter(AdminRight.class)
-    @SecuredAction(value = "", type = ActionType.AUTHENTICATED)
+    @SecuredAction(value = "", type = ActionType.RESOURCE)
+    @ResourceFilter(AccessIfMyStructure.class)
```
Comment (lapeyronnie) [inline src/main/java/fr/openent/viescolaire/controller/EleveController.java:194]: Ne pas changer le ressource filter. Par contre, changer le type d'action RESSOURCE à la place d'AUTHENTICATED

**Review by lapeyronnie — CHANGES_REQUESTED** [review]
changer type d'action

File: src/main/java/fr/openent/viescolaire/controller/EleveController.java [hunk 3]
```diff
@@ -213,8 +190,8 @@ public void getAppreciationDevoirEleve(final HttpServerRequest request) {
 
     @Get("/structures/:structureId/students")
     @ApiDoc("Get list of students from a structure")
+    @SecuredAction(value = "", type = ActionType.RESOURCE)
     @ResourceFilter(AdminRight.class)
```
Comment (simonchaval) [inline src/main/java/fr/openent/viescolaire/controller/EleveController.java:194]: Need to have viescolaire access view to use this API

**Review by lapeyronnie — APPROVED** [review]

**Pull request merged**